module/VuFind/tests/integration-tests/src/VuFindTest/Mink/BasicSearchTest.php

```diff
@@ -58,4 +58,126 @@ public function testOutOfBoundsPage()
             trim($this->findCss($page, '.search-stats')->getText())
         );
     }
+
+    /**
+     * Test simple top pagination
+     *
+     * @return void
+     */
+    public function testSimpleTopPagination(): void
+    {
+        $session = $this->getMinkSession();
+        $session->visit($this->getVuFindUrl() . '/Search/Results');
+        $page = $session->getPage();
+
+        // No paginator unless configured:
+        $this->unFindCss($page, '.search-header .pagination');
+        $this->unFindCss($page, '.search-header .pagination-simple');
+
+        // Enable pagination:
+        $this->changeConfigs(
+            [
+                'searches' => [
+                    'General' => [
+                        'top_paginator' => 'simple',
+                    ],
+                ],
+            ]
+        );
+
+        $session->visit($this->getVuFindUrl() . '/Search/Results');
+        $this->assertStringContainsString(
+            "Showing 1 - 20 results",
+            $this->findCss($page, '.search-stats')->getText()
+        );
+
+        // No prev page on first page:
+        $this->unFindCss($page, '.search-header .pagination-simple .page-prev');
+
+        $secondPage = $this->findCss($page, '.search-header .pagination-simple .page-next');
+        $secondPage->click();
+        $this->waitForPageLoad($page);
+        $this->assertStringContainsString(
+            "Showing 21 - 40 results",
+            $this->findCss($page, '.search-stats')->getText()
+        );
+        // Prev page now present, click it:
+        $this->clickCss($page, '.search-header .pagination-simple .page-prev');
+        $this->waitForPageLoad($page);
+        $this->assertStringContainsString(
+            "Showing 1 - 20 results",
+            $this->findCss($page, '.search-stats')->getText()
+        );
+    }
+
+    /**
+     * Test full top pagination
+     *
+     * @return void
+     */
+    public function testFullTopPagination(): void
+    {
+        $session = $this->getMinkSession();
+        $session->visit($this->getVuFindUrl() . '/Search/Results');
+        $page = $session->getPage();
+
+        // No paginator unless configured:
+        $this->unFindCss($page, '.search-header .pagination');
```

Comment:
Why are you checking for `.search-header .pagination` here but using `.pagination-top` when you actually interact with the page? It seems like it might be worth making the selectors more consistent (or at least adding additional find/unfind checks) to make it more clear that the test is proving what it says that it's proving.